### ISSUE TITLE
Grails 3.0.8: CommandObject binding null to 'null' String

### ISSUE THREAD

**Issue by todoubaba** [grails/grails-core#9331]
When binding `request.JSON` to a command object in the controller, the `null` value will turn into `"null"` string.
```
{"x": null}
       |
       v
def cmd = new MyCmd(request.JSON)
       |
       v
assert(cmd.x == 'null')
```
It works fine in grails 3.0.4.

**Comment by jeffbrown** [comment]
It looks like there is an inconsistency here and I will investigate but you can probably get around the problem by following more common practice.  Since you are creating the instance your self, that isn't really command object from Grails' perspective.  What Grails calls a command object is the thing that the framework instantiates for you and passes into your controller actions.

    class SomeController {
        def someAction(MyCmd cmd) {
            // cmd is a command object
        }
    }

One problem with `new MyCmd(request.JSON)` is that you are invoking the default `Map` constructor which will not use Grails' data binding system.  If you really want to create the instance yourself and you want to use the data binding system then you have some options, including something like this...

    class SomeController {
        def someAction() {
            def cmd = new MyCmd()
            bindData cmd, request.JSON
        }
    }

I think with either of those approaches, cmd.x will be `null` not `'null'`.

All of this is related to https://github.com/grails/grails-core/issues/9162 and https://github.com/grails/grails-core/issues/9129.  A relevant Spec is at https://github.com/grails/grails-core/blob/master/grails-plugin-converters/src/test/groovy/grails/converters/ParsingNullJsonValuesSpec.groovy.

Thanks for the feedback.

**Comment by jeffbrown** [comment]
The issue here has to do with the fact that the Grails data binder is not being used when you invoke the constructor of a non-domain constructor.  The default Groovy constructor behavior is in place and the `toString()` method of `org.grails.web.json.JSONObject.Null` is being used, which returns the String `'null'`.  Changing that is potentially problematic.  See #9162 and #9129.

The problematic scenario for Grails 3.0.x which is described above is one like this...


```
class SomeController {
    def someAction() {
        def cmd = new MyCmd(request.JSON)

        // the rest of your code here
    }
}
```

The following would all work...

```
class SomeController {
    def someAction(MyCmd cmd) {
        // the rest of your code here
    }
}
```

```
class SomeController {
    def someAction() {
        def cmd = new MyCmd()
        bindData cmd, request.JSON

        // the rest of your code here
    }
}
```

```
class SomeController {
    def someAction() {
        def cmd = new MyCmd()
        bindData cmd, request

        // the rest of your code here
    }
}
```

I think we should close this as Won't Fix for 3.0.x.  The desired behavior does exist in Grails 3.1 and beyond.
